recipes/lely-core/all/conanfile.py

```diff
@@ -0,0 +1,238 @@
+    topics = ("canopen",)
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "rt": [True, False],
+        "threads": [True, False],
+        "ecss-compliance": [True, False],
+        "errno": [True, False],
+        "malloc": [True, False],
+        "stdio": [True, False],
+        "cxx": [True, False],
+        "daemon": [True, False],
+        "diag": [True, False],
+        "canfd": [True, False],
+        "dcf": [True, False],
+        "dcf-restore": [True, False],
+        "obj-default": [True, False],
+        "obj-file": [True, False],
+        "obj-limits": [True, False],
+        "obj-name": [True, False],
+        "obj-upload": [True, False],
+        "sdev": [True, False],
+        "csdo": [True, False],
+        "rpdo": [True, False],
+        "tpdo": [True, False],
+        "mpdo": [True, False],
+        "sync": [True, False],
+        "time": [True, False],
+        "emcy": [True, False],
+        "lss": [True, False],
+        "wtm": [True, False],
+        "master": [True, False],
+        "ng": [True, False],
+        "nmt-boot": [True, False],
+        "nmt-cfg": [True, False],
+        "gw": [True, False],
+        "gw-txt": [True, False],
+        "coapp-master": [True, False],
+        "coapp-slave": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "rt": True,
+        "threads": True,
+        "ecss-compliance": False,
+        "errno": True,
+        "malloc": True,
+        "stdio": True,
+        "cxx": True,
+        "daemon": True,
+        "diag": True,
+        "canfd": True,
+        "dcf": True,
+        "dcf-restore": True,
+        "obj-default": True,
+        "obj-file": True,
+        "obj-limits": True,
+        "obj-name": True,
+        "obj-upload": True,
+        "sdev": True,
+        "csdo": True,
+        "rpdo": True,
+        "tpdo": True,
+        "mpdo": True,
+        "sync": True,
+        "time": True,
+        "emcy": True,
+        "lss": True,
+        "wtm": True,
+        "master": True,
+        "ng": True,
+        "nmt-boot": True,
+        "nmt-cfg": True,
+        "gw": True,
+        "gw-txt": True,
+        "coapp-master": True,
+        "coapp-slave": True,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def validate(self):
+        # Windows requires proprietary from https://www.ixxat.com/technical-support/support/windows-driver-software
+        # hence support for it will be skipped for now
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(
+                "This Conan version of this package is only compatible with Linux"
```

Comment:
Can you add some context to the error here? (Like the comment above) UX would be nicer in case someone triggers this locally